### ISSUE TITLE
Package builds fine with stack but not on hackage

### ISSUE THREAD

**Issue by locallycompact** [roman/Haskell-etc#43]
Hi, I'm using this as a dependency in a project on hackage but it's failing to build the executable. I guess it's something to do with the flags in stack.yaml not reaching hpack? Is it possible to declare these in hpack directly?

Here's the error: http://hackage.haskell.org/package/baserock-schema-0.0.3.4/reports/3

**Comment by roman** [comment]
Indeed, the problem is that the `cli` cabal flag is not true, you need to specify the cabal flags to whatever is building your project, I don't know if you can specify the cabal flags on the `.cabal` file itself

**Comment by locallycompact** [comment]
The .cabal file is autogenerated by hpack, I'm not sure how to specify these flags in the .package file.

**Comment by roman** [comment]
Who is calling the cabal command? I know you can specify cabal flags via CLI

**Comment by locallycompact** [comment]
I have no idea what calls the cabal command on hackage, it doesn't seem to say in the build log.

**Comment by locallycompact** [comment]
Is it possible to release a version that has these flags on by default? I don't really understand the use case for having these features disabled.

**Comment by locallycompact** [comment]
I've asked on freenode and been told it isn't possible to express a dependency on a package built with a specific flag, so it doesn't seem like hackage will be able to build projects using etc while the feature flags are disabled by default.

**Comment by roman** [comment]
> Is it possible to release a version that has these flags on by default? I don't understand the use case for having these features disabled.

I'm sorry for the troubles @locallycompact; I did add the cabal flags to reduce the footprint of dependencies.

I'll add tickets to re-implement this feature strategy by having different projects:

* `etc-cli`
* `etc-yaml`
* `etc-utils` 

These changes will require some refactoring of the internals of the project. I think those are necessary to move forward with adding new backends (AWS parameter-store comes to mind).

Thanks for using the library, and apologies for choosing Cabal Flags for this functionality :-(
